packages/accounts-base/accounts_server.js

```diff
@@ -1135,13 +1135,15 @@ Ap._expirePasswordResetTokens = function (oldestValidDate, userId) {
   var userFilter = userId ? {_id: userId} : {};
 
   this.users.update(_.extend(userFilter, {
+    "services.password.reset.reason": reason,
     $or: [
       { "services.password.reset.when": { $lt: oldestValidDate } },
       { "services.password.reset.when": { $lt: +oldestValidDate } }
     ]
   }), {
     $unset: {
       "services.password.reset": {
```

Comment:
This `$unset` looks to me in generally strange. Is this really a valid MongoDB query? Strange.